avro-schema-processor/src/main/java/io/micronaut/avro/visitor/AvroSchemaVisitor.java

```diff
@@ -0,0 +1,398 @@
+                avroSchema.setJavaClass(type.getCanonicalName());
+            }
+            default -> avroSchema.setType(Type.DOUBLE);
+        }
+    }
+
+    private static void setBeanSchemaFields(ClassElement element, VisitorContext visitorContext, AvroSchemaContext context, AvroSchema avroSchema) {
+        avroSchema.setType(Type.RECORD);
+
+        avroSchema.setNamespace(element.getPackageName());
+        context.currentOriginatingElements().add(element);
+        if (avroSchema.getName() == null) {
+            avroSchema.setName(element.getSimpleName().replace('$', '.'));
+        }
+        context.createdSchemasByType().put(element.getName(), avroSchema);
+
+        for (PropertyElement property : element.getBeanProperties()) {
+            AvroSchema.Field field = new AvroSchema.Field();
+            field.setName(property.getName());
+
+            if (property.hasAnnotation(Avro.class)) {
+                AnnotationValue<Avro> fieldSchema = property.getAnnotation(Avro.class);
+                String[] aliases = fieldSchema.stringValues("aliases");
+                if (aliases.length > 0) {
+                    field.setAliases(Arrays.asList(aliases));
+                }
+                fieldSchema.stringValue("doc").ifPresent(field::setDoc);
+            }
+
+            // Create schema for the property
+            AvroSchema propertySchema = createSchema(property, visitorContext, context);
+            if (isPrimitiveType(property.getType()) || isPrimitiveAvroType(propertySchema.getType()) && !isLogicalAvroType(propertySchema.getLogicalType()) && !propertySchema.isUnsupported() || propertySchema.isRefType()) {
+                field.setType(propertySchema.getType());
+            } else {
+                field.setType(propertySchema);
+            }
+
+            avroSchema.addField(field);
+        }
+    }
+
+    private static void writeSchema(AvroSchema avroSchema, ClassElement originatingElement, VisitorContext visitorContext, AvroSchemaContext context) {
+        String fileName = getFileName(avroSchema);
+        String path = context.outputLocation() + SLASH + fileName;
+        GeneratedFile specFile = visitorContext.visitMetaInfFile(path, originatingElement).orElse(null);
+        if (specFile == null) {
+            visitorContext.warn("Unable to get [\" " + path + "\"] file to write Avro schema", null);
+        } else {
+            visitorContext.info("Generating Avro schema file: " + specFile.getName());
+            try (Writer writer = specFile.openWriter()) {
+                ObjectMapper mapper = ObjectMapper.getDefault();;
+                List<AvroSchema.Field> sortedFields = avroSchema.getFields().stream()
+                    .sorted(Comparator.comparing(AvroSchema.Field::getName))
+                    .collect(Collectors.toList());
+                avroSchema.setFields(sortedFields);
+                String schema = mapper.writeValueAsString(avroSchema);
```

Comment:
Use [writeValue](https://docs.micronaut.io/latest/api/io/micronaut/json/JsonMapper.html#writeValue(java.io.OutputStream,java.lang.Object)) instead of buffering in the string